internal/impl/devbox.go

```diff
@@ -827,16 +830,22 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	nixEnvPath := env["PATH"]
 	debug.Log("PATH after plugins and config is: %s", nixEnvPath)
 
-	env["PATH"] = JoinPathLists(nixEnvPath, originalPath)
+	if pure {
+		env["PATH"] = nixEnvPath
+	} else {
+		env["PATH"] = JoinPathLists(nixEnvPath, originalPath)
+	}
```

Comment:
I think this if-else branching is not needed, because originalPath on line 744-748 will be empty-string. This could be as before IIUC.